pkg/client/interface.go

```diff
@@ -0,0 +1,36 @@
+package client
+
+import (
+	"github.com/filecoin-project/mir/pkg/pb/requestpb"
+	t "github.com/filecoin-project/mir/pkg/types"
+)
+
+// The Client represents an SMR client that produces new transactions (using NewTX)
+// that can be submitted to the SMR system.
+// It sequentially assigns transaction numbers to all created transactions.
+// The Client is informed (using Done)
+// when a transaction is considered applied (or otherwise safe to stop caring about).
+// All created transactions that have not been confirmed this way are considered pending.
+// The client is expected to persistently store all pending transactions,
+// in case they need to be resubmitted after recovering from a potential crash.
+type Client interface {
```

Comment:
Can we rename the interface or package to something that would describe the behavior?
Otherwise, we will have `client.Client` in the code. Even `smr.Client` sounds better. Is that correct?

Reply:
I'd like it to be a separate package though, specific to the client, and both names (package and type) should be descriptive. The standard library does that too (`time.Time`). I wouldn't say it's a big deal, so I'd suggest leaving it, unless you have strong objections...